src/types/FunctionType.ts

```diff
@@ -38,20 +38,20 @@ export class FunctionType implements BscType {
         return this;
     }
 
-    public isAssignableTo(targetType: BscType) {
+    public isAssignableTo(targetType: BscType, context?: TypeContext) {
```

Comment:
Will our new `TypeExpression` concept negate the need for this new `TypeContext` param?

Reply:
No ... TypeContext is used to find out "What context is this type referenced from?" .. for example, a file included in `ComponentA` might have a class defined a certain way, but that same file included in a different component might have a different definition for the same class name, just based on other files included in that scope.